src/azure-cli-core/azure/cli/core/commands/__init__.py

```diff
@@ -284,9 +287,23 @@ def _resolve_default_value_from_config_file(self, arg, overrides):
         self._add_vscode_extension_metadata(arg, overrides)
 
         # same blunt mechanism like we handled id-parts, for create command, no name default
-        if self.name.split()[-1] == 'create' and overrides.settings.get('metavar', None) == 'NAME':
-            return
-        super(AzCliCommand, self)._resolve_default_value_from_config_file(arg, overrides)
+        if not (self.name.split()[-1] == 'create' and overrides.settings.get('metavar', None) == 'NAME'):
+            super(AzCliCommand, self)._resolve_default_value_from_config_file(arg, overrides)
+
+        self._resolve_default_value_from_local_context(arg, overrides)
+
+    def _resolve_default_value_from_local_context(self, arg, overrides):
+        if self.cli_ctx.local_context.is_on():
+            lca = overrides.settings.get('local_context_attribute', None)
+            if not lca or not lca.actions or USE not in lca.actions:
+                return
+            if lca.name:
+                local_context = self.cli_ctx.local_context
+                value = local_context.get(self.name, lca.name)
+                if value:
+                    logger.debug("local context '%s' for arg %s", value, arg.name)
+                    overrides.settings['default'] = DefaultStr(value)
+                    overrides.settings['required'] = False
```

Comment:
>overrides.settings['required'] = False [](start = 20, length = 38)

will this statement "overrides.settings['required'] = False" impact the ux like -h ?

Reply:
Good point. Yes, if the value is saved in local context, it will show the value as default in -h. I think we need add more information for this, such as `from local context`.
And if local context is not turned on, -h will behavior as before.